### DISCUSSION TITLE
Conversation about config management (at the example of tf)

### DISCUSSION THREAD

**Discussion by blaggacao** [terranix/terranix#71] [General]
Hi @mrVanDalo !

I've been using terranix extensively at work. In parallel, I've prototyped and am exploring somewhat of a "simpler", not nixpkgs (!) module system based approach to config mgt, including terraform.

- The prototype impl can be found here (in case interest is sufficient): https://github.com/divnix/data-merge/
- This strategy also depends on a json-validator to enforce some rudimentary "compile time" (eval time) static checks before any configuration ships to the remote api's advanced, maybe stateful, validation. See a WIP impl here: https://github.com/NixOS/nix/pull/5403

If the interest is sufficient, I'd like to deepen discussion on (alternative, hopefully simpler) approches to config mgt, including tf.

**Comment by blaggacao** [comment]
If not interested, please just close. I'm aware that this bug tracker might not be the most welcomed venue for kicking this off. Sorry for the bothering, then.

**Comment by mrVanDalo** [comment]
Hey @blaggacao, 

I have problems understanding what this is about. I would replace the NixOS module system but only with something that is even more powerful. Do you want to provide an additional module system, if so I'm happy to integrate it in terranix.

I'm also bad with abbreviations, cfg? (I guess configuration) mgt? sic? have no meaning for me.

> I've been using terranix extensively at work. 

Nice, that's good to hear :D 

> If not interested, please just close. I'm aware that this bug tracker might not be the most welcomed venue for kicking this off. Sorry for the bothering, then.

No worries, I'm happy to discuss this topic, but as I said, I have problems understanding what this is about.
